platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/BasicSoftwareVersion.java

```diff
@@ -108,29 +108,9 @@ public int compareTo(final SoftwareVersion that) {
         }
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public boolean equals(final Object o) {
-        if (this == o) {
-            return true;
-        }
-
-        if (!(o instanceof BasicSoftwareVersion that)) {
```

Comment:
Would it be better to implement it and throw an exception?

Reply:
This is intended to be used by non-hedera applications, so I think throwing is the wrong thing to do. It would make sense to do this if this class was not accessible outside of test code.